datastacks/README.md

```diff
@@ -0,0 +1,84 @@
+# Datastacks
+
+[Datastacks](https://github.com/amido/stacks-azure-data/tree/main/Datastacks) is a Python library
+built to support the Ensono Stacks Data Platform which can be used for various functions within the project.
+
+At present the library works to ease usage of the templates within the [de_templates](https://github.com/amido/stacks-azure-data/tree/main/de_templates) directory, and provides a cli which can be used to generate new components to be used within a data platform.
+
+## Key components
+
+The key components of datastacks currently include:
+
+- **Generate**: This command contains subcommands which generate components for the data platform given a config file. 
+
+  - **Ingest**: This command utilises the template for ingest data pipelines, and uses a given config file to generate the required code for a new ingest pipeline ready for use. A flag can be included to specify whether or not to include data quality components in the pipeline
+
+### Using CLI
+
+```bash
+datastacks --help
+datastacks generate --help
+datastacks generate ingest --help
+datastacks generate ingest --config-path="de_templates/test_config_ingest.yaml"
+datastacks generate ingest --config-path="de_templates/test_config_ingest.yaml" --data-quality
+```
+
+### Using an entrypoint script
+
+```bash
+python datastacks_cli.py --help
+python datastacks_cli.py generate --help
+python datastacks_cli.py generate ingest --help
+python datastacks_cli.py generate ingest --config-path="de_templates/test_config_ingest.yaml"
+python datastacks_cli.py generate ingest --config-path="de_templates/test_config_ingest.yaml" --data-quality
+```
+
+## Required Config File
+
+For component generation, the Datastacks cli take a path to a config file. This config file should be a yaml file and have the below format. A sample config file is included in the templates folder.
+
+```yaml
```

Comment:
Could you include the yaml here by referencing that file instead of a direct copy?

Reply:
Can do, I just thought it was useful to have in line aswell, but I'm not attached to that idea

Reply:
It should be possible to add it in a way that it's rendered inline without actually copying and pasting the text itself.